src/ngRoute/route.js

```diff
@@ -160,7 +171,7 @@ function $RouteProvider() {
    */
   this.when = function(path, route) {
     //copy original route object to preserve params inherited from proto chain
-    var routeCopy = angular.copy(route);
+    var routeCopy = createShallowCopy(route);
```

Comment:
We could, but I wouldn't.

We don't need the exact same function. The one from core can shallow-copy arrays as well, ignores `$$` prefixed properties and supports passing your own `dst` object, which properties are shallow-copied onto. We we don't need any of that.